pkg/machine/qemu/machine.go

```diff
@@ -579,7 +584,10 @@ func (v *MachineVM) Start(name string, _ machine.StartOptions) error {
 	if err != nil {
 		return err
 	}
-
+	conn.Close()
+	dnr.Close()
+	dnw.Close()
+	fd.Close()
```

Comment:
These should be `defer fd.Close()` after they are opened. Right now this code will leak the open fds when an error occurs and it returns early.